### ISSUE TITLE
KeyError in _internal_cap_handler

### ISSUE THREAD

**Issue by hemebond** [snoonetIRC/async-irc#5]
Attempting to do a little test. Not sure if this is a bug in async-irc or my usage.
```python
# main.py
import asyncio

from asyncirc.protocol import IrcProtocol
from asyncirc.server import Server

servers = [
	Server("irc.freenode.net", 6667),
]

async def log(conn, message):
	print(message)

async def main():
	conn = IrcProtocol(servers, "BotNick", loop=loop)
	conn.register_cap('userhost-in-names')
	conn.register('*', log)
	await conn.connect()
	await asyncio.sleep(24 * 60 * 60)

loop = asyncio.get_event_loop()

try:
	loop.run_until_complete(main())
except KeyboardInterrupt:
	pass
finally:
	loop.stop()
```

```
$ python ./main.py 
:Delphini.GA.US.StarChat.Net NOTICE * :*** Looking up your hostname...
:Delphini.GA.US.StarChat.Net NOTICE * :*** Couldn't resolve your hostname; using your IP address instead
:Delphini.GA.US.StarChat.Net CAP * LS :unrealircd.org/plaintext-policy=user=allow,oper=warn,server=deny unrealircd.org/link-security=1 extended-join chghost cap-notify userhost-in-names multi-prefix away-notify account-notify sasl=EXTERNAL,PLAIN tls
PING :85762D26
Task exception was never retrieved
future: <Task finished coro=<_internal_cap_handler() done, defined at /lib/python3.6/site-packages/asyncirc/protocol.py:40> exception=KeyError('',)>
Traceback (most recent call last):
  File "/lib/python3.6/site-packages/asyncirc/protocol.py", line 59, in _internal_cap_handler
    current = conn.server.caps[cap.name][0]
KeyError: ''
:Delphini.GA.US.StarChat.Net CAP BotNick ACK :userhost-in-names 
:Delphini.GA.US.StarChat.Net CAP BotNick ACK :sasl 
Task exception was never retrieved
future: <Task finished coro=<_internal_cap_handler() done, defined at /lib/python3.6/site-packages/asyncirc/protocol.py:40> exception=KeyError('',)>
Traceback (most recent call last):
  File "/lib/python3.6/site-packages/asyncirc/protocol.py", line 59, in _internal_cap_handler
    current = conn.server.caps[cap.name][0]
KeyError: ''
```

```
$ pip freeze
async-irc==0.1.7
py-irclib==0.1.4
```

For some reason the caplist has two elements in it, `sasl` and an empty/blank cap. I worked around it will a try/except:
```python
            try:
                current = conn.server.caps[cap.name][0]
                conn.server.caps[cap.name] = (current, enabled)
                if enabled:
                    handlers = filter(None, conn.cap_handlers[cap.name])
                    await asyncio.gather(*[func(conn, cap) for func in handlers])
            except KeyError:
                pass
```

**Comment by linuxdaemon** [comment]
Hm, this is an odd one. There should not be any empty entries in caplist.

**Comment by linuxdaemon** [comment]
The issue appears to be in py-irclib, the CAP ACK from freenode has an extra space at the end and the lib doesn't strip the text. I'll push an update to the lib and add some more test cases

**Comment by linuxdaemon** [comment]
Ok, the issue should be fixed, update `py-irclib` to `0.1.5`

**Comment by hemebond** [comment]
That appears to have resolved the issue. Thanks.
